src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -284,6 +284,22 @@
         "DELTA",  # https://github.com/delta-io/delta
         "XML",  # https://github.com/databricks/spark-xml
     ),
+    # Adding Hint related segments so they are not treated as generic comments
+    # https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-hints.html
+    StartHintCommentSegment=StringParser(
+        "/*+", KeywordSegment, name="start_hint_comment"
+    ),
+    EndHintCommentSegment=StringParser("*/", KeywordSegment, name="end_hint_comment"),
```

Comment:
This isn't going to work. If you run `sqlfluff parse` on any of the below examples you'll see they are still being parsed as comments. This is because the Lexer grabs anything beginning with `/*` and ending with `*/` as a `block_comment` (you can see this in the ANSI dialect that Spark3 inherits from).

To allow these to be parsed separately you need to add the below code to "steal" these sections before they get swallowed up by the comment lexer:

```py
spark3_dialect.insert_lexer_matchers(
    [
        RegexLexer("start_hint", r"\/\*\+", CodeSegment),
    ],
    before="block_comment",
)

spark3_dialect.insert_lexer_matchers(
    [
        RegexLexer("end_hint", r"\*\/", CodeSegment),
    ],
    before="single_quote",
)
```